### PR TITLE
Update houdini license validator

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
As reported in this [community comment](https://community.ynput.io/t/ayon-monthly-beginners-topic-november-23/967/9)
Houdini USD publishing is only restricted in Houdini apprentice. 

## Testing notes:
1. publish USD from an apprentice Houdini license won't work in OP or Ayon.

### CODE REVIEW

File: openpype/hosts/houdini/plugins/publish/validate_houdini_license_category.py [hunk 1]
```diff
@@ -3,30 +3,29 @@
...
-    """Validate the Houdini instance runs a Commercial license.
+class ValidateHoudiniNotApprenticeLicense(pyblish.api.InstancePlugin):
+    """Validate the Houdini instance runs a non Apprentice license.
 
-    When extracting USD files from a non-commercial Houdini license, even with
-    Houdini Indie license, the resulting files will get "scrambled" with
-    a license protection and get a special .usdnc or .usdlc suffix.
+    When extracting USD files from an apprentice Houdini license,
+    the resulting files will get "scrambled" with a license protection
+    and get a special .usdnc or .usdlc suffix.
```
Comment (BigRoy) [inline openpype/hosts/houdini/plugins/publish/validate_houdini_license_category.py:11]:
```suggestion
    and get a special .usdnc suffix.
```

The `.usdlc` extension was from the previous Houdini Indie restrictions over three years ago (basically the name refers to: usd lesser commercial). That restriction has been lifted.

**Review by BigRoy — APPROVED** [review]
Other than the docstring comment this looks good to me.

**Comment by moonyuet** [comment]
I actually think that it would be great to add this before the abc extraction from pointcache as alembic export doesn't support in apprentice version

**Comment by MustafaJafar** [comment]
> I actually think that it would be great to add this before the abc extraction from pointcache as alembic export doesn't support in apprentice version

Totally good point!

**Comment by MustafaJafar** [comment]
any ideas why `instance.data["families"]` doesn't contain `instance.data["family"]` value ?
I was trying `self.log.debug(instance.data["families"])` in license validator 
but I can't find `pointcache` nor `usd` in their respective logs ? 
image

**Comment by BigRoy** [comment]
> any ideas why `instance.data["families"]` doesn't contain `instance.data["family"]` value ? I was trying `self.log.debug(instance.data["families"])` in license validator but I can't find `pointcache` nor `usd` in their respective logs ? image
> 
> I can find that line here but it's skipped
> 
> https://github.com/ynput/OpenPype/blob/d34ac5637e1d8b9c832309efb22fd79d0a94fb2b/openpype/hosts/houdini/plugins/publish/collect_instances.py#L80

That plugin does nothing anymore since new publisher, see [here](https://github.com/ynput/OpenPype/blob/d34ac5637e1d8b9c832309efb22fd79d0a94fb2b/openpype/hosts/houdini/plugins/publish/collect_instances.py#L50-L53) and can basically be removed.

> any ideas why instance.data["families"] doesn't contain instance.data["family"] value ?

Because unfortunately that's how the system works. There's no hard requirement currently that `families` will contain `family` itself. I'd prefer if it automatically would but it doesn't by design in pyblish - as such there's no requirement for `families` to even exist or match `family` value. There can be cases thus where it's undefined or does not contain `family` itself.

File: openpype/hosts/houdini/plugins/publish/validate_houdini_license_category.py [hunk 2]
```diff
@@ -3,30 +3,29 @@
...
         import hou
 
-        license = hou.licenseCategory()
-        if license != hou.licenseCategoryType.Commercial:
+        if hou.isApprentice():
             raise PublishValidationError(
-                ("USD Publishing requires a full Commercial "
-                 "license. You are on: {}").format(license),
+                ("USD Publishing requires a non apprentice "
+                 "license."),
```
Comment (BigRoy) [inline openpype/hosts/houdini/plugins/publish/validate_houdini_license_category.py:30]: I believe these brackets are redundant, right?
```suggestion
                "USD Publishing requires a non apprentice "
                "license.",
```

However, I even think you're not passing the line length either:
```suggestion
                "USD Publishing requires a non apprentice license.",
```

**Comment by MustafaJafar** [comment]
> `families = ["usd", "abc"]` and it should then run on relevant plug-ins just fine.
>
> Or what issues are you facing?

the issue with the printed message 
using `instance.data["family"]` returns `pointcache` or `usd` 
using `instance.data["families"]` returns `[abc]` or `[ ]` 
so, I was asking if there's a solution other than using both of them .. 
e.g. is there a way to retrieve the creator label using `creator_identifier` ? 

---

A test of my last commits: 

image

File: openpype/hosts/houdini/plugins/publish/validate_houdini_license_category.py [hunk 3]
```diff
@@ -1,32 +1,37 @@
...
-    label = "Houdini Commercial License"
+    label = "Houdini Apprentice License"
 
     def process(self, instance):
 
-        import hou
+        if hou.isApprentice():
+            families = [instance.data["family"]]
+            families += instance.data.get("families", [])
+            families = " ".join(families).title()
```
Comment (BigRoy) [inline openpype/hosts/houdini/plugins/publish/validate_houdini_license_category.py:33]:
```suggestion
            # Find which family was matched with the plug-in
            families = {instance.data["family"]}
            families.update(instance.data.get("families", []))
            disallowed_families = families.intersection(self.families)
            families = " ".join(sorted(disallowed_families)).title()
```

Comment (MustafaJafar) [reply openpype/hosts/houdini/plugins/publish/validate_houdini_license_category.py:33]: both solutions work
image

Comment (BigRoy) [reply openpype/hosts/houdini/plugins/publish/validate_houdini_license_category.py:33]: Great.
Make sure to lowercase the "P" on "Publishing" by the way.

Comment (MustafaJafar) [reply openpype/hosts/houdini/plugins/publish/validate_houdini_license_category.py:33]: what about ? 

```suggestion
            # Find which family was matched with the plug-in
            families = {instance.data["family"]}
            families.update(instance.data.get("families", []))
            families = " ".join(families).title()
```
image

Comment (BigRoy) [reply openpype/hosts/houdini/plugins/publish/validate_houdini_license_category.py:33]: You still have `Pointcache` in your logs in that last example - which is what we're trying to exclude.

Comment (MustafaJafar) [reply openpype/hosts/houdini/plugins/publish/validate_houdini_license_category.py:33]: I missed that `Pointcache` is mentioned in the error label hehe :D

Comment (BigRoy) [reply openpype/hosts/houdini/plugins/publish/validate_houdini_license_category.py]: Unresolving this as the issue still stands - and my code snippet shared above should still be a valid fix.

Comment (MustafaJafar) [reply openpype/hosts/houdini/plugins/publish/validate_houdini_license_category.py]: Thanks!

**Review by BigRoy — APPROVED** [review]
Didn't test, but code looks good to me.

**Review by antirotor — APPROVED** [review]
Looks good!

**Pull request merged**